Assets/Scripts/Models/Time/TimeManager.cs

```diff
@@ -130,6 +134,16 @@ public void Update(float time)
         // Systems that update every frame.
         InvokeEvent(EveryFrame, time);
 
+        if (nextFrameActions.Count > 0)
+        {
+            for (int i = 0; i < nextFrameActions.Count; i++)
```

Comment:
Alright then, that explains that. I had much "fun" in Java in the last couple of months, where it's actually generally faster to do foreach-style `for` loops than iterative `for` loops. (Yes, they have the same keyword for both, I loathe it.)

Anyway, good to know that the old Mono seems to be bad in that regard. Maybe it will be better in Unity 5.5 with newer Mono version, but who knows.